volatility3/framework/plugins/windows/scheduled_tasks.py

```diff
@@ -0,0 +1,1796 @@
+    """
+    Enumeration that maps trigger types to their magic number encodings
+    """
+
+    WindowsNotificationFacility = 0x6666
+    Session = 0x7777
+    Registration = 0x8888
+    Logon = 0xAAAA
+    Event = 0xCCCC
+    Time = 0xDDDD
+    Idle = 0xEEEE
+    Boot = 0xFFFF
+
+
+class Weekday(enum.Enum):
+    """
+    Enumeration that contains bitwise values for days of the week.
+    """
+
+    Sunday = 0x1
+    Monday = 0x2
+    Tuesday = 0x4
+    Wednesday = 0x8
+    Thursday = 0x10
+    Friday = 0x20
+    Saturday = 0x40
+
+
+class Months(enum.Enum):
+    """
+    Enumeration that contains bitwise values for months of the year.
+    """
+
+    January = 0x1
+    February = 0x2
+    March = 0x4
+    April = 0x8
+    May = 0x10
+    June = 0x20
+    July = 0x40
+    August = 0x80
+    September = 0x100
+    October = 0x200
+    November = 0x400
+    December = 0x800
+
+
+class SidType(enum.Enum):
+    """
+    Enumeration that maps SID types to their encoded integer values
+    """
+
+    User = 1
+    Group = 2
+    Domain = 3
+    Alias = 4
+    WellKnownGroup = 5
+    DeletedAccount = 6
+    Invalid = 7
+    Unknown = 8
+    Computer = 9
+    Label = 10
+    LogonSession = 11
+
+
+@dataclasses.dataclass
+class TaskSchedulerTime:
+    """
+    A class containing datetime information about when a task will run
+    """
+
+    is_localized: bool
+    filetime: Optional[datetime.datetime]
+
+
+@dataclasses.dataclass
+class TaskSchedulerTimePeriod:
+    """
+    Class containing information delimiting time periods within scheduled tasks.
+    """
+
+    years: int
+    months: int
+    weeks: int
+    days: int
+    hours: int
+    minutes: int
+    seconds: int
+
+
+JOB_BUCKET_FLAGS = {
+    0x2: "Run only if idle",
+    0x4: "Restart on idle",
+    0x8: "Stop on idle end",
+    0x10: "Disallow start if on batteries",
+    0x20: "Stop if going on batteries",
+    0x40: "Start when available",
+    0x80: "Run only if network available",
+    0x100: "Allow start on demand",
+    0x200: "Wake to run",
+    0x400: "Execute parallel",
+    0x800: "Execute stop existing",
+    0x1000: "Execute queue",
+    0x2000: "Execute ignore new",
+    0x4000: "Logon type s4u",
+    0x10000: "Logon type InteractiveToken",
+    0x40000: "Logon type Password",
+    0x80000: "Logon type InteractiveTokenOrPassword",
+    0x400000: "Enabled",
+    0x800000: "Hidden",
+    0x1000000: "Runlevel highest available",
+    0x2000000: "Task",
+    0x4000000: "Version",
+    0x8000000: "Token SID type none",
+    0x10000000: "Token SID type unrestricted",
+    0x20000000: "Interval",
+    0x40000000: "Allow hard terminate",
+}
+
+NULL = "\u0000"
+
+
+class _ScheduledTasksReader(io.BytesIO):
+
+    def read_task_scheduler_time(self) -> Optional[Tuple[bool, datetime.datetime]]:
+        is_localized = bool(self.read_aligned_u1())
+        filetime = self.decode_filetime()
+        if filetime is None:
+            return None
+        return is_localized, filetime
```

Comment:
If we know whether it's localized, could we not just return a python datetime object that includes the localization, rather than a filetime?  The user might want the value changed to a timezone of their choosing, so keeping everything in one format we understand seems better than a separate flag (which appears to be ignored later on when this is called)?

Reply:
The `is_localized` field is really there for the Windows task scheduler UI to display the 'synchronize across time zones' checkbox and the datetime (converted to local datetime, if the box is unchecked, or as UTC otherwise). In the binary format, the timestamps are all UTC, so this isn't important for determining when something will actually run.

Reply:
Ok, so... do we still need this information or could it be encoded in the timestamp?  Just wonder if it's worth handing back a tuple?

Reply:
There's not really anything useful we can do with it now, I'll remove it from the return value.